ember/app/ui/protected/route.js

```diff
@@ -3,22 +3,10 @@ import { inject as service } from "@ember/service";
 
 export default class ProtectedRoute extends Route {
   @service session;
-  @service store;
 
-  beforeModel(transition) {
-    this.session.requireAuthentication(transition, "login");
-  }
+  async beforeModel(transition) {
+    await this.session.requireAuthentication(transition, "login");
 
-  model() {
-    return Promise.all([
-      this.store.query(
-        "membership",
-        {
-          include: "organisation",
-        },
-        { adapterOptions: { customEndpoint: "my-memberships" } }
-      ),
-      this.store.queryRecord("identity", {}),
-    ]);
+    await this.session.currentIdentity
```

Comment:
Is this working out? I always thought we have to await the task instance like this `await this.session.fetchCurrentIdentity.lastRunning`, but reading trough the documentation raises some doubt on my beliefs... :exploding_head:

Reply:
if we stop the code there and call `currentIdentity.toString()` we get `"<Task:fetchCurrentIdentity> TaskInstance" `